omnigibson/sensors/vision_sensor.py

```diff
@@ -447,18 +459,22 @@ def _register_instance(self, instance_name, id=False):
         if instance_name not in registry.values():
             registry[len(registry)] = instance_name
 
-    def _remap_bounding_box_semantic_ids(self, bboxes):
+    def _remap_bounding_box_semantic_ids(self, bboxes, id_to_labels):
         """
         Remap the semantic IDs of the bounding boxes to our own semantic IDs.
 
         Args:
             bboxes (list of dict): List of bounding boxes to remap
+            id_to_labels (dict): Dictionary of semantic IDs to class labels
         Returns:
             list of dict: Remapped list of bounding boxes
+            dict: Remapped id_to_labels dictionary
         """
+        replicator_mapping = self._preprocess_semantic_labels(id_to_labels)
         for bbox in bboxes:
-            bbox["semanticId"] = VisionSensor.SEMANTIC_REMAPPER.remap_bbox(bbox["semanticId"], self._scene)
-        return bboxes
+            bbox["semanticId"] = semantic_class_name_to_id()[replicator_mapping[bbox["semanticId"]]]
```

Comment:
seems wasteful to call `semantic_class_name_to_id()` every time? Why not use another remapper instead?

Reply:
Potentially we could! Here are some counterarguments though:
1. We don't really have many bounding boxes per observation
2. This `semantic_class_name_to_id()` is actually a cached function that returns a dictionary, which means each look up would be constant time without function call overhead. Compared to mapping 128x128/624x624/1024x1024 images, this is probably nothing.

Reply:
sounds good